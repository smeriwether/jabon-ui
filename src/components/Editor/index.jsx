import React from 'react';
import PropTypes from 'prop-types';
import AceEditor from 'react-ace';
import 'brace/mode/xml';
import 'brace/theme/tomorrow';
import 'brace/ext/language_tools';

const propTypes = {
  value: PropTypes.string,
  name: PropTypes.string.isRequired,
  readOnly: PropTypes.bool,
  onChange: PropTypes.func
};

const defaultProps = {
  readOnly: false,
  onChange: function() {}
};

const editorOptions = {
  enableBasicAutocompletion: true,
  enableLiveAutocompletion: true,
  showLineNumbers: true,
  tabSize: 2
};

const Editor = ({ value, name, readOnly, onChange }) => {
  return (
    <AceEditor
      mode="xml"
      theme="tomorrow"
      name={ name }
      height="100%"
      width="100%"
      value={ value }
      readOnly={ readOnly }
      showPrintMargin={ false }
      showGutter={ true }
      highlightActiveLine={ true }
      setOptions={ editorOptions }
      onChange={ onChange }
    />
  );
};

Editor.propTypes = propTypes;
Editor.defaultProps = defaultProps;

export default Editor;
