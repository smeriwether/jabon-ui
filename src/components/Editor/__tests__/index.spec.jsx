import Editor from '../../Editor';

describe('<Editor />', () => {
  const setup = (overrides) => {
    const props = {
      value: null,
      name: null,
      onChange: jest.fn(),
      ...overrides
    };

    const wrapper = shallow(<Editor { ...props } />);

    return { wrapper };
  };

  it('should match the snapshot', () => {
    const { wrapper } = setup({ value: '<foo></foo>', name: 'editor' });
    expect(wrapper).toMatchSnapshot();
  });
});
