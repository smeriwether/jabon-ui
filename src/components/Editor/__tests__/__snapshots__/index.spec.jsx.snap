// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Editor /> should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Editor
    name="editor"
    onChange={[MockFunction]}
    readOnly={false}
    value="<foo></foo>"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "cursorStart": 1,
      "editorProps": Object {},
      "enableBasicAutocompletion": false,
      "enableLiveAutocompletion": false,
      "focus": false,
      "fontSize": 12,
      "height": "100%",
      "highlightActiveLine": true,
      "maxLines": null,
      "minLines": null,
      "mode": "xml",
      "name": "editor",
      "onChange": [MockFunction],
      "onLoad": null,
      "onPaste": null,
      "onScroll": null,
      "readOnly": false,
      "scrollMargin": Array [
        0,
        0,
        0,
        0,
      ],
      "setOptions": Object {
        "enableBasicAutocompletion": true,
        "enableLiveAutocompletion": true,
        "showLineNumbers": true,
        "tabSize": 2,
      },
      "showGutter": true,
      "showPrintMargin": false,
      "style": Object {},
      "tabSize": 4,
      "theme": "tomorrow",
      "value": "<foo></foo>",
      "width": "100%",
      "wrapEnabled": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "cursorStart": 1,
        "editorProps": Object {},
        "enableBasicAutocompletion": false,
        "enableLiveAutocompletion": false,
        "focus": false,
        "fontSize": 12,
        "height": "100%",
        "highlightActiveLine": true,
        "maxLines": null,
        "minLines": null,
        "mode": "xml",
        "name": "editor",
        "onChange": [MockFunction],
        "onLoad": null,
        "onPaste": null,
        "onScroll": null,
        "readOnly": false,
        "scrollMargin": Array [
          0,
          0,
          0,
          0,
        ],
        "setOptions": Object {
          "enableBasicAutocompletion": true,
          "enableLiveAutocompletion": true,
          "showLineNumbers": true,
          "tabSize": 2,
        },
        "showGutter": true,
        "showPrintMargin": false,
        "style": Object {},
        "tabSize": 4,
        "theme": "tomorrow",
        "value": "<foo></foo>",
        "width": "100%",
        "wrapEnabled": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
